js/common/util/compute_domain.js

```diff
@@ -0,0 +1,9 @@
+import { extractDomain } from 'home-assistant-js-websocket';
+
+export default function computeDomain(stateObj) {
+  if (!stateObj._domain) {
```

Comment:
I sometimes wonder if this is not just a stupid optimization and just calling `extractDomain` each time would be faster.